### DISCUSSION TITLE
Job submission gui for Windows and Linux (python/tk based)

### DISCUSSION THREAD

**Discussion by PaulAltair** [OpenRadioss/OpenRadioss#1231] [Show and tell]
Important news, as of 26th September 2024, the gui submission tool has reached a level of maturity that it is now migrated from a confluence download to be included in the official Tools repository:

[https://github.com/OpenRadioss/Tools/tree/main/openradioss_gui](https://github.com/OpenRadioss/Tools/tree/main/openradioss_gui)

Since the original post in march 2023, many improvements have been made, see below for latest gui appearance: and download and contribute in the tools repository

Thanks again for the contributions of @hotaosa to this tool

![OpenRadioss_gui](https://github.com/user-attachments/assets/c489a10f-07b4-47a8-948d-ffae910f077e)

-----------------------------

To help make job submission easier (especially for the precompiled windows version) I have written a small, basic python/tk based job submission gui, Windows and Linux versions are available

![image](https://user-images.githubusercontent.com/98418895/225296435-011483b6-c50f-4b59-8808-0945536cae10.png)

Download and instructions for use are available at confluence pages:
[python/tk guis in confluence](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609)

There are also python based launchers to run the Anim-vtk and T01-csv conversion tools

----------------------------------------

**Comment by PaulAltair** [comment]
Linux version now updated to support mpi submissions

![image](https://user-images.githubusercontent.com/98418895/225379065-cdd13d31-11f4-4fb0-925f-74cdcfbcca81.png)

**Reply by PaulAltair** [reply]
version now available supporting mpi under windows if installed

**Comment by ZdendaH** [comment]
Hello, Thank you for the work you do. I found it but I do not know how to use it. I double-click on anim-vtk.bat and the window shows just milliseconds and disappeared. The same happened with a file for submitting the job. I tried to run the Python script file in Visual Studio Code but I did not have success. I do not know what I am doing wrong.

<img width="686" alt="image" src="https://user-images.githubusercontent.com/115239640/227355003-af68afaa-654f-4e28-9e4d-f46f2f6562fe.png">

Does anyone have an idea of what could help to solve it?
Thank you for your reply.

**Reply by PaulAltair** [reply]
For windows execution, there are now exes available that don't require python install as pre-requisite, the instructions for use are here [GUI Job Submission Guide in Confluence](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609/python+tk+guis+for+job+submission+anim-vtk+conversion+and+T01-csv+conversion) Hope this helps

**Comment by PaulAltair** [comment]
FYI I today have uploaded v2 of the job submission guis, now supporting:

Multiple Engine Files (e.g. 0001.rad, 0002.rad, 0003.rad etc) select all in Engine Selection at runtime
Running 'only' Engine File(s) (for restarts, e.g. launch a 0002.rad and restart an existing run) Select only engine file(s)
Running 'only' Starter (if you want to check the Starter only), still creates .rst file(s) Select only starter file

The ability to run .k or .key (Dyna Format models) remains in place, and there is a version allowing selection of a vars.bat to assist mpi running under windows.

Let me know if you have any issues in your application, or any other ideas for improvement

Available from confluence pages:
[Gui submission tools at confluence](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609/python+tk+guis+for+job+submission+anim-vtk+conversion+and+T01-csv+conversion)

**Reply by bthomas84** [reply]
Hi I encounter an issue. Trying to launch a radioss run that already ran on the commercial radioss, i have an error :
***  ERROR : Unknown command line argument: days
Since i don't write the command line myself thanks to the GUI I don't really understand where does it come from.

Kind regards

**Reply by PaulAltair** [reply]
Is there a space in your install path or filename? 

The robustness of the gui when it/OpenRadioss are installed in a directory with spaces is not guaranteed!

Possibly it is trying to read part of the submission filename/path as an argument to the script? Try renaming file without space?

Paul

**Reply by bthomas84** [reply]
You're completely right... space in the directory name, thank you Paul!

**Reply by PaulAltair** [reply]
Is this under windows or linux btw... I should try and code something better for this when I get some time, but it isn't that simple due to the pushd/popd the script is doing to get the files in/out of the submission folder...

**Reply by bthomas84** [reply]
It's on Windows, don't worry a simple change of directory plays the trick. Thank you vey much for your so quick answer

**Comment by hotaosa** [comment]
Hello. I have written a gui(for Windows) supporting for STOP/KILL control, multiple engine files and queue of jobs. I referred to your script. I hope it helps you to imporove your gui.

[openradioss_run_script_for_windows_mpi.zip](https://github.com/OpenRadioss/OpenRadioss/files/11975231/openradioss_run_script_for_windows_mpi.zip)

Please modify line 18 of "openradioss_run_script.bat" to set MPI_PATH
Multiple jobs can be queued and will be executed in order
If the next engine file(ex.0002.rad) exist in the same directory, it will executed automatically
When you submit starter file(0000.rad), the result files are deleted if they exist in the same directory

**Comment by PaulAltair** [comment]
Really nice, thank you! I like the sub windows and queuing a lot, its a lot neater than mine! :) I will take a deeper look and try to combine our efforts and port it to linux version. In the meantime, may I share your gui 'as is' also on the confluence page? [Job submission guis page at OpenRadioss confluence](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609/python+tk+guis+for+job+submission+anim-vtk+conversion+and+T01-csv+conversion)

**Reply by hotaosa** [reply]
Thank you for your reply. Or course it's okay! I'm glad you like it!

**Reply by PaulAltair** [reply]
I have updated the page at confluence to include your tool too, thanks again!

[Confluence link](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609/python+tk+guis+for+job+submission+anim-vtk+conversion+and+T01-csv+conversion)

**Reply by PaulAltair** [reply]
The attached is largely your code @hotaosa so thank you again!, I've made some modifications based on my scripts but the framework is yours not my old ones since it was so much neater!:

Updates:
Accepts .k or .key files as input in addition to .rad files
I modified the size of the console window so the output no longer has to wrap
I added buttons to the console window to request ANIM or H3D updates
I added OpenRadioss icons to all the windows in place of the standard tk feather
There is a linux port version (assumes open mpi is installed and doesn't need a vars file) 

[OpenRadiossScriptsRevised.zip](https://github.com/OpenRadioss/OpenRadioss/files/11996704/OpenRadiossScriptsRevised.zip)

**Reply by hotaosa** [reply]
I will utilize the tool you have improved, thank you!

**Reply by PaulAltair** [reply]
Final version... though sure to be more bug fixes to come!: **EDITED already :)**

BugFix to last zip: Dyna input files were getting incorrect name in jobwindow title (fixed)
                             Some files were not deleted in run directory in Linux (fixed)
                             Made file search in windows file delete slightly more stringent (minor risk unintended files could be deleted, fixed)

Updates:
Options added to auto convert Anim-Vtk and/or T01-csv at end of run with checkboxes at submission.
Option to run single precision version with checkbox.
If np<=1 on linux and windows, smp engine is run (mpi vars.bat is not required for windows)
If np>=2 on first attempt Windows version prompts to browse to vars.bat, or use np=1, doesn't prompt once set
Job queue update to show, vtk, csv, sp/dp
**+1** Window 'x/close' on main panel performs same function as close
**+2** Taskbar icon for windows now populates with openradioss logo
Think that is it...

[Updated_guis.zip](https://github.com/OpenRadioss/OpenRadioss/files/12021986/Updated_guis.zip)

I'll give us a week or so further bug testing this before I promote it more widely and upload to confluence

Thanks again @hotaosa for your valuable input without which this wouldn't exist!

**Reply by hotaosa** [reply]
Checkboxes are very nice!
I have a (probably final) suggestion: How about adding a feature to remember states of the checkboxes?
I modified it to save the states to 'config' directory when a job added, and load it at startup. I hope you'll like it.
[win_scripts_mk3_mod.zip](https://github.com/OpenRadioss/OpenRadioss/files/12022548/win_scripts_mk3_mod.zip)



P.S. apply_config() in my code may be better this way:
```
def apply_config():
    if os.path.exists('hoge'):
        with open('hoge', mode='r') as f:
            if f.readline() == 'True':
                foo_status.set(True)
```

**Reply by PaulAltair** [reply]
Great, thanks, yes! I have updated with this suggestion :)
I was able to create desktop (and toolbar, with pin to toolbar) shortcuts now for windows too btw: maybe you already did this? 

Create Windows Shortcut with form e.g.
Target: C:\Windows\pyw.exe "D:\OpenRadioss\win_scripts_mk3\openradioss_run_script_for_windows_mpi_ps.py"
Start in: D:\OpenRadioss\win_scripts_mk3\
Change icon to the one in the script icon folder: D:\OpenRadioss\win_scripts_mk3\icon\ross.ico
![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/6975b3b8-5779-423f-a5ad-ff72fd6d09bd)

**Reply by hotaosa** [reply]
I didn't create shortcuts, so it was very helpful. Thank you!

**Reply by PaulAltair** [reply]
and for linux: to have desktop icon:
add/create a file: ./local/share/applications/openradioss.desktop (make executable)
content of form e.g.:
```
[Desktop Entry]
Type=Application
Name=OpenRadioss
Exec=bash -c "cd /home/psharp/OpenRadioss/OpenRadioss-PaulTesting/linux_scripts_mk3 && /usr/bin/python3 /home/psharp/OpenRadioss/OpenRadioss-PaulTesting/linux_scripts_mk3/openradioss_run_script_for_linux_mpi_ps.py"
Icon=/home/psharp/OpenRadioss/OpenRadioss-PaulTesting/linux_scripts_mk3/icon/ross.png
```
This will add to menus: then can pin to desktop also:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/e7295001-d7d9-4fd1-8be7-7ef9aa408d9b)

**Reply by PaulAltair** [reply]
With your savestate mod, and dropdown for info

[Updated_guis_120723.zip](https://github.com/OpenRadioss/OpenRadioss/files/12029329/Updated_guis_120723.zip)

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/117cdecf-6909-4ecf-a4f8-18de8fad934c)

**Reply by PaulAltair** [reply]
minor update: now handles situation where user submits a starter without an engine more cleanly (issues message 'no engines found' and terminates)
[Updated_guis_190723.zip](https://github.com/OpenRadioss/OpenRadioss/files/12096556/Updated_guis_190723.zip)

**Reply by ZdendaH** [reply]
Hi Paul, hotaosa, ...

I wanted to run calculation by using GUI. It looks good. Thanks for that. I ran the calculation on one core, it worked. If I use mpi options I get the message on the picture

<img width="656" alt="image" src="https://github.com/OpenRadioss/OpenRadioss/assets/115239640/49ff6567-c36c-4691-bea6-91773ecb4836">

"PLEASE CHECK LISTING FILE FOR FURTHER DETAILS
/Intel/oneAPI/mpi/2021.10.0/env/vars.bat was unexpected at this time."
   
I have file "path_to_mpi_vars.txt" where the folder to vars.bat" is specified
<img width="877" alt="image" src="https://github.com/OpenRadioss/OpenRadioss/assets/115239640/110c2f49-ac3c-4a05-8f2a-4cd182b728d3">

I tried to modify the openradioss_run_script_for_windows_mpi_ps.py file to add the file path instead the file name.
<img width="810" alt="image" src="https://github.com/OpenRadioss/OpenRadioss/assets/115239640/dd5dafb8-4b6c-4160-94fe-fbf6c13ee024">
It is not working neither file.txt nor specified file in python script.

Do you have idea what could be the way to work? Maybe I have bad SLAH/BACKSLASH in file path. I use windows 11. I do not know. 
Thank you.

**Reply by hotaosa** [reply]
Hello, ZdendaH.
It seems that the path shown in the job window is incorrect. I think it is because there is a space in the path. Enclosing the path with double quotations might work. Sorry if you've tried it.

**Reply by PaulAltair** [reply]
Yes, I think that will be it, I'll try to modify the file selector for the vars.bat to account for paths with spaces!

**Reply by PaulAltair** [reply]
Ok, I've verified, if you enclose the path in your 'path_to_mpi_vars.txt' in quotes ("") it should work ok,

I've updated the windows py to account for this now by modifying the vars.bat file selector to account for vars.bat in path with spaces, the updated version is attached here, or, you can update line 91 of the windows py script to file.write('"' + file_path + '"')

[UpdatedGuis230723.zip](https://github.com/OpenRadioss/OpenRadioss/files/12138568/UpdatedGuis230723.zip)

**Reply by ZdendaH** [reply]
Thank you for modification. Now It works very well.

**Reply by PaulAltair** [reply]
one further update added to the tools, now the queue window allows user to manually start the next or last job (runs in another job window, additionally to the queue being processed, can be used to run multiple jobs at same time)

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/ffc110ff-41d7-4ae8-ab20-26930779964f)

[UpdatedGuis240723.zip](https://github.com/OpenRadioss/OpenRadioss/files/12140304/UpdatedGuis240723.zip)

**Reply by hotaosa** [reply]
Thank you for the nice feature!

I found a issue. When multiple jobs within the same directory are executed at the same time, 'running_job' file is overwritten. As a result, using the STOP button may cause unintended job to stop. 
It might be better to write 'running_job_"job_base_name"' based on the job's name to avoid this problem. (However, this change might require modifications in many places.)

**Reply by PaulAltair** [reply]
hmm, ok, thanks for the catch, I hadn't considered jobs being run in the same folder... I'll give it some thought tomorrow!

**Reply by PaulAltair** [reply]
think this fixes it, windows only, will sort out linux tomorrow: 
[win_scripts_mk3.zip](https://github.com/OpenRadioss/OpenRadioss/files/12140983/win_scripts_mk3.zip)

**Reply by PaulAltair** [reply]
ok, actually, fixed linux too, and recreated win exe, confluence page zips are updated, or here:
[UpdatedGuis240723b.zip](https://github.com/OpenRadioss/OpenRadioss/files/12141028/UpdatedGuis240723b.zip)

**Reply by PaulAltair** [reply]
I will tidy up some more tomorrow... the feature isn't needed for starter phase, and I think I can code something that removes the need for the 'running_job' file completely

**Reply by PaulAltair** [reply]
Ok, I couldn't get rid of the status file(s), but, I gave the stop, kill, anim, h3d buttons different behaviour during starter since none of them would work there (Stop/Kill will cause job to end at end of starter, Anim, h3d give message), and since I'd implemented that and I could reuse the status check, I added 'starter only' checkbox to the main submission window, this allows user to run starter only even if engines are present. Dropdown now includes link to confluence page too.
[UpdatedGuis240723c.zip](https://github.com/OpenRadioss/OpenRadioss/files/12147476/UpdatedGuis240723c.zip)

**Comment by PaulAltair** [comment]
I have now updated the confluence page reflecting the latest updates: [Confluence guis page](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/45252609/python+tk+guis+for+job+submission+anim-vtk+conversion+and+T01-csv+conversion)

**Reply by PaulAltair** [reply]
@hotaosa I will publicise this more widely via linkedin etc next week, I will credit you by your github username, if you also wish to be credited in person (linked in profile etc) let me know!

**Reply by hotaosa** [reply]
Thank you for offering to credit me. I don’t need any more specific credit. I'm looking forward to it!

**Reply by PaulAltair** [reply]
I posted to LinkedIn today, https://www.linkedin.com/posts/paul-sharp-08a4a436_opensourcecommunity-opensource-openradioss-activity-7089596719952211968-2Ry_?utm_source=share&utm_medium=member_android

**Comment by MzHf0** [comment]
Hello, 
I am trying to use the gui for job submission but always get the message:

  _ERROR : 
  Reader configuration files are not available
  Check and set RAD_CFG_PATH variable.
  Standard configuration file installation is 
  %ALTAIR_HOME%\hwsolvers\radioss\cfg_

Can anyone help me with that?
Tank you.

**Reply by hotaosa** [reply]
Hello. It seems the path to the solver directory is not set up correctly. The script folder should be extracted the installation folder of OpenRadioss. (or you need to modify "openradioss_run_script.bat".) Please refer to here: https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/58785793/Latest+gui+scripts

**Reply by MzHf0** [reply]
Thank you for your quick answer! I just figured out the problem. Somehow some files where not unziped correctly. A new download and unzipping of Open Radioss solved the problem.

**Reply by JasGarnier** [reply]
I got this error too. It seemed to be caused by having the script files folder in the openradioss folder. I moved it elsewhere and explicitly set the path via "set OPENRADIOSS_PATH=C:\OpenRadioss" and the error disappeared.

**Comment by ser2177** [comment]
Hello,
I am attempting to utilize the latest gui script for submitting job applications, but I consistently receive the following message:

'findstr' is not recognized as an internal or external command,

operable program or batch file.

**Reply by hotaosa** [reply]
Hello. findstr.exe is usually exists in C:\Windows\System32. Could you check this directory is set in the 'Path' environment variable?

**Reply by ser2177** [reply]
Hello, path variable is defined.

**Reply by PaulAltair** [reply]
If your path variable includes C:\Windows\System32 then I am not sure why you get this error, it should be present in all windows installs afaik
![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/5e13dd57-a34f-483e-a3e8-460d98666fd4)

**Reply by ser2177** [reply]
There is a findstr.exe file in the C:\Windows\System32 folder.

**Reply by hotaosa** [reply]
I'm not sure too.
How about replacing all findstr in openradioss_run_script_ps.bat with "C:\Windows\System32\findstr.exe"?

**Reply by ser2177** [reply]
There are no error messages, but radioss doesn't work.

**Reply by PaulAltair** [reply]
Strange, going back to basics, can you confirm that you have placed the gui submissions folder inside your openradioss installation (same level as exec) that OpenRadioss is installed in a path with no spaces in it, and that you are trying to run a filename with no spaces from a directory/path with no spaces? 

Does a commandline submission work ok?

**Reply by ser2177** [reply]
I agree with you. I will uninstall OpenRadioss and perform a clean installation. I have changed too many parameters. There is a possibility that it is not working due to one of them.

I can run it from the command line.

**Reply by PaulAltair** [reply]
if it works from commandline, it ought not be necessary to uninstall/reinstall, but it is worth a go...

when you state that there are no errors but radioss doesn't work, what does happen? do you get the console pop up and no content? or the console never launches?

**Reply by JasGarnier** [reply]
I got that too. Solved it by adding PATH=C:\Windows\System32 in the Windows environment variables. Normally I'd have thought that would be there already but some software installs might screw it up.

**Comment by ser2177** [comment]
The console opens as shown in the image below.
![console](https://github.com/OpenRadioss/OpenRadioss/assets/126450107/2748e7c7-8f67-4030-a9a0-0f627972fff6)

**Reply by PaulAltair** [reply]
Ok, not sure what the issue is behind this still then. Looks like the python/tk bits are working ok, but there is still an issue with the bat script for your setup.

I'm out of office today, but tomorrow I will look into it further and see if we can narrow down the cause by running the bat from commandline outside of the gui.

Thanks for your patience

**Comment by ser2177** [comment]
Thank you for your interest. Today, I performed a clean installation and added some commands to the batch file as system paths. I also believe that there are no issues with Python/tk. However, after the clean installation, when I run it from the command line, I encounter a new error.

![image](https://github.com/OpenRadioss/OpenRadioss/assets/126450107/3cf41d83-f0cf-4af0-9120-ec187065ec17)

**Comment by ser2177** [comment]
I deleted the line “set OPENRADIOSS_PATH=%ABS_PATH%” and defined it on the system. It It works in both SMP and MPP modes. Thank you for your assistance.

**Reply by PaulAltair** [reply]
Ah, ok, thanks for the update, that line would normally pick up the OpenRadioss install path based on the location of the run script folder (inside OpenRadioss install) not sure why that wouldn't work for you, but glad you have a workaround!

**Comment by GNSamuelGao** [comment]
When I using this gui in Centos7, there is error message, as below. Could you help to figure out how to solve this error? I already install python3-tk, or not installed right?
Traceback (most recent call last):
  File "openradioss_run_script_for_linux_mpi_ps.py", line 18, in <module>
    icon_image = tk.PhotoImage(file='./icon/ross.png')
  File "/usr/lib64/python3.6/tkinter/__init__.py", line 3545, in __init__
    Image.__init__(self, 'photo', name, cnf, master, **kw)
  File "/usr/lib64/python3.6/tkinter/__init__.py", line 3501, in __init__
    self.tk.call(('image', 'create', imgtype, name,) + options)
_tkinter.TclError: couldn't recognize data in image file "./icon/ross.png"

**Comment by martijnt1** [comment]
Seems to be centOS7 lacking PNG support, have you installed libraries [libpng-dev] and [zlib], or tried converting the PNG into a JPG + updateling the icon extension in the script?



Mar 20, 2024, 02:58 by ***@***.***:

**Reply by GNSamuelGao** [reply]
Thanks martijnt1. I have installed many libraries related png, jpg... But it's still error. I have remove all the image code, and it can run now.  Without any good appearance, but it works at least. :)

**Comment by GNSamuelGao** [comment]
Another issue is how to use MPI in Linux. I cound not find the OneAPI in OpenRadioss folder. As below code quote:

export PATH=$OPENRADIOSS_PATH/extlib/intelOneAPI_runtime/linux64:$PATH

**Reply by MarianBulla** [reply]
Hi  @GNSamuelGao,
on Linux, you can also use OpenMPI. I'm showing it here: https://www.youtube.com/watch?v=ddvH2CNfaYg&t=466s
But you should also be able to use IntelMPI. 
I did not test it on Linux, but on windows here: https://www.youtube.com/watch?v=Uo5miT_a-YY&t=393s

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by martijnt1** [comment]
Glad its running, w your workaround.Not knowing if you tried the 2nd possible solution mentioned, to adjust the PNG image+code for a JPG.
Confirming Marian's feedback, MPI is running at least on Debian & Ubuntu, unable to comment on CentOS (RHEL) yet.

Mar 21, 2024, 07:33 by ***@***.***:

**Comment by GNSamuelGao** [comment]
Hi @martijnt1 & @MarianBulla ,

Thanks both. I am a new hand for Github. I have tried different way, but I still have below questions
1. For icon issue, I have tried both PNG and JPG and installed many lib, but it still can not recognize the image file. But it can work at least~
          __Error:__ _tkinter.TclError: couldn't recognize data in image file "./icon/ross.jpg" .
![gui](https://github.com/OpenRadioss/OpenRadioss/assets/162549477/7f18fc61-9ae9-43dc-a335-f1ba0ed3bb07)

2. For MPI, maybe I need to build the "engine_linux64_gf_ompi" by myself? Because from the latest GUI code, it needs intelOneApi lib in the exec folder, see pic, but I can't find the lib in exec folder.
![path](https://github.com/OpenRadioss/OpenRadioss/assets/162549477/b55b5016-f5c5-4a0c-819f-34bf7a146f9e)
I also try to install oneapi, but there is also error: cannot find libmpi.so.40. I check my resource, only libmpi.so.12
![MPI Error2](https://github.com/OpenRadioss/OpenRadioss/assets/162549477/9e70b8bf-d838-4ded-ad8a-6714cd0ef021)

Br
Thanks

**Reply by PaulAltair** [reply]
Sorry, I had missed this reply, apologies for the lengthy time to respond. I think in my ubuntu install (where I wrote the gui), I installed ompi from apt-get rather than compiling it, I should update it to use a compiled ompi per the install documentation. I'll look into it.

**Reply by PaulAltair** [reply]
The installation for OPENMPI under linux is decribed in the [HOWTO.md](https://github.com/OpenRadioss/OpenRadioss/blob/main/HOWTO.md)

in the openradioss_run_script_ps.sh can you then try adding these 2 lines? it is working for me then with self compiled open mpi per the HOWTO

 `export LD_LIBRARY_PATH=/opt/openmpi/lib:$LD_LIBRARY_PATH`
`export PATH=/opt/openmpi/bin:$PATH`

**Reply by GNSamuelGao** [reply]
Hi Paul,

Thanks for your reply. I have tried again by following HowTo and you advice. It works now. Thanks. 
But here come another issue is the GUI is not fit to Win11. Which parameter I can change to change the GUI to show all text. I can change the GUI width but not height.
![Snipaste_2024-07-11_10-14-08](https://github.com/OpenRadioss/OpenRadioss/assets/162549477/6873f505-d052-4327-b75e-47fde39461d3)

**Reply by PaulAltair** [reply]
ok, I use win11 myself, and it is ok here, so perhaps it is down to graphics scaling settings (do you have text scaling set higher than 100%?) 

I just tried setting mine at e.g. 150% and can replicate the issue you have

the size of the dialog is set in the file 'openradioss_run_script_for_windows_mpi_ps.py' lines 18-20

![image](https://github.com/user-attachments/assets/7480dc72-c583-4ed0-85cb-e4a5377564b8)

you can change the values there or set 'root.resizable' as True, True to allow you to change the height

you can also change the size of the buttons if you need to in the same file on lines 183, 185, 187, 189

![image](https://github.com/user-attachments/assets/433e8a6d-fd91-476e-a34f-40ed2aa46be5)

I'm not sure how to do a more general fix for robustness on all installs, I found some info about importing ctypes for DPI awareness in tk, but they don't seem to help here, I'll keep looking into it!

**Reply by GNSamuelGao** [reply]
Thanks for the detail info. It's useful.

**Comment by PaulAltair** [comment]
With the great initiative of the Vortex-Radioss anim-d3plot converter: [Discussion 2361](https://github.com/orgs/OpenRadioss/discussions/2361) I have now have updated the job submission gui: available here: [Job Submission guis](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/58785793/Latest+gui+scripts)

So that there is now an option to run the Vortex-Radioss d3plot conversion at the end of a run if it is installed

Let me know if you have any difficulties

**Reply by Vortex-CAE** [reply]
Thank you Paul! This is a great development to help improve workflow!

**Reply by PaulAltair** [reply]
Based on feedback, I made some small updates to the packages today, both linux and windows version include a helper d3convert.py script

**Comment by hdrz** [comment]
Hi,

I started using the run script on linux, and it runs fine. Just a small suggestion, maybe add a shebang line at the top of the script, so double clicking it (or directly run it from the commandline) wil work:

`#!/usr/bin/env python3`

This should work on all linux distributions.

Thanks!

**Reply by PaulAltair** [reply]
Thanks, I have added this in the latest (mk5) version

**Comment by PaulAltair** [comment]
Latest Mk5 versions available, the gui now detects if Vortex-Radioss python module is present or not, and presents the gui with/without d3plot conversion options as appropriate. 

Also, if present, a d3plot button is added to run window, allowing conversion of any anim files present in the run directory while the job is running

![image](https://github.com/user-attachments/assets/d5d700b0-7944-42b1-a55f-d5bb054dbfba)

**Reply by PaulAltair** [reply]
As of September 26th 2024, The gui is now migrated to the official OpenRadioss tools repository, downloads and contributions should be done there from now on [https://github.com/OpenRadioss/Tools/tree/main/openradioss_gui](https://github.com/OpenRadioss/Tools/tree/main/openradioss_gui)

![OpenRadioss_gui](https://github.com/user-attachments/assets/7cff1d38-f35b-4507-b88f-84e8ca8cf6ed)

**Comment by hoanpqh** [comment]
Hello, I can not see Kill, Stop, Anim button. Please show me how to solve this problem!

**Reply by PaulAltair** [reply]
Are you using the current build in tools repository? In Linux or Windows? Does resizing the window make them visible? (they disappear if you make the window smaller)

**Reply by PaulAltair** [reply]
I just submitted a request to stop this happening (buttons should remain visible on resizing)

**Reply by hoanpqh** [reply]
> Are you using the current build in tools repository? In Linux or Windows? Does resizing the window make them visible? (they disappear if you make the window smaller)

Hello, I use the current build in tools repository in Linux. 
Resizing the window does not make them visible!

**Comment by PaulAltair** [comment]
I just tested current build in my Lubuntu Linux and it is working ok for me there: What linux version are you using, maybe it is due to graphics/desktop implementation there, does the gui otherwise work ok? e.g. queue window, submission window buttons?
![image](https://github.com/user-attachments/assets/143bc776-0360-4b72-ae97-e3580c90f862)

**Reply by hoanpqh** [reply]
I am using Ubuntu 20 in a Virtual Machine. I attach the screenshot. TKS
![VirtualBox_Ubuntu_17_01_2025_08_56_37](https://github.com/user-attachments/assets/a5cadd37-7881-41f3-957b-968876f900e7)

**Reply by hoanpqh** [reply]
I increased display solution and now I can see the buttons. By the way, I can not open the GUI by running OpenRadioss_gui.bash. I have to run python3 OpenRadioss_gui.py instead.

**Reply by PaulAltair** [reply]
what happens when you try to run the bash?

**Reply by PaulAltair** [reply]
I think if you take the latest build from tools the modification I made the other day should work even at the lower resolution btw

**Reply by hoanpqh** [reply]
> what happens when you try to run the bash?

It works now. Thank you very much.
![VirtualBox_Ubuntu_17_01_2025_10_09_17](https://github.com/user-attachments/assets/836d4e34-6999-4944-929a-88401e3312e2)

**Reply by PaulAltair** [reply]
Great, your screenshot still shows the older version though!

The newer release should be in next stable build release too, so if you have this one working now then you can wait for that (It changes check boxes to drop-down menu as well as change to window buttons)
